Tests/ReactiveSwiftTests/FoundationExtensionsSpec.swift

```diff
@@ -103,15 +110,15 @@ class FoundationExtensionsSpec: QuickSpec {
 
 				expect(DispatchTimeInterval.milliseconds(500).timeInterval).to(beCloseTo(0.5))
 				expect(DispatchTimeInterval.milliseconds(250).timeInterval).to(beCloseTo(0.25))
-				expect(DispatchTimeInterval.never.timeInterval) == Double.infinity
+				expect(DispatchTimeInterval.never.timeInterval).to(equal(Double.infinity))
 			}
 
 			it("should negate as you'd hope") {
 				expect((-DispatchTimeInterval.seconds(1)).timeInterval).to(beCloseTo(-1.0))
 				expect((-DispatchTimeInterval.milliseconds(1)).timeInterval).to(beCloseTo(-0.001))
 				expect((-DispatchTimeInterval.microseconds(1)).timeInterval).to(beCloseTo(-0.000001, within: 0.0000001))
 				expect((-DispatchTimeInterval.nanoseconds(1)).timeInterval).to(beCloseTo(-0.000000001, within: 0.0000000001))
-				expect((-DispatchTimeInterval.never).timeInterval) == Double.infinity
```

Comment:
@andersio also if you don't mind I suggest changes to 3 particular test case expressions within this file. From my point of view, `expect.to.equal` syntax explicitly expresses expectation of equality when `==` operator makes people get inside Nimble and look for necessary overload to make sure whether expression is not false-positive.

Reply:
I thought this is `Equatable` based, and the stdlib floating point types already imply IEEE 1754 equivalence.

Reply:
Sure, the idea is following:

Using any kind of spec + expectation/matchers framework: (Kiwi, Specta/Expecta, Quick/Nimble) I expect the following pseudo-syntax line for test assertion:
```
expect(inspectable-expression) -> kind-of-expectation(to/notTo, eventually, eventuallyBeforeTimingOutAfter..) -> matcher-expression
```
All of mentioned above frameworks provide us with some DSL to set test expectations, so as a developer I expect an expectation (sorry for tautology) to be set using DSL. With Expecta I can write `expect(someInt) == 0`, and it won't warn me because I can compare returned object with 0 in Objective-C. But the test flow will be wrong, because it won't verify this against any matcher. Instead I should write `expect(someInt).to.equal(0)`, which is correct in terms of its DSL.

Seeing `expect((-DispatchTimeInterval.never).timeInterval) == Double.infinity` in Swift makes me find whether its valid DSL syntax first. I get to https://github.com/Quick/Nimble/blob/master/Sources/Nimble/Matchers/Equal.swift#L172 and find that there's an implementation of `==` operator which makes `lhs.to(equal(rhs)` that I expected in the first place.

Maybe my confusion is connected with unsafe Objective-C background, where equality comparisons could be done mistakingly with reference arithmetics, and we should disregard it, however I would still wonder why to use operator `==` over matcher syntax for particular equality case. Especially, if this operator is implemented by calling matcher syntax.

Reply:
`==` are the fluent conveniences of the matchers that are meant to help reduce the verbosity AFAIK. It could be confusing for Swift newcomers with C family language background, but IMO it does make sense given equality and inequality in Swift are implemented as operators, with compile tim overloading & type safety guarding for the problems you may otherwise encounter.

Reply:
No problem, bringing `==` syntax back. I don't have that strong opinion regarding this readability case, so knowing that's just a convenience matcher makes me fine with this approach. Also, I've checked Nimble matchers README and they show using operator `==` for equality matcher and `===` for identity matcher. Thus this point might be considered resolved?